Src/FluentAssertions/Equivalency/INode.cs

```diff
@@ -22,9 +24,18 @@ public interface INode
     string Name { get; set; }
 
     /// <summary>
-    /// Gets the type of this node.
+    /// Gets the type of this node, e.g. the type of the field or property, or the type of the collection item. 
     /// </summary>
     Type Type { get; }
+    
+    /// <summary>
+    /// Gets the type of the parent node, e.g. the type that declares a property or field.
+    /// </summary>
+    /// <value>
+    /// Is <c>null</c> for the root object.
+    /// </value>
+    [CanBeNull]
+    Type ParentType { get; }
```

Comment:
We should be explicit about whether `ParentType` is the compile-time or runtime type of the parent.

Reply:
It's the `DeclaringType`, so that shouldn't matter.

Reply:
Ohh.. it's the declaring type.
From the name I got the impression that it was either the compile-time or runtime type of the parent _instance_ carrying the value.

How about naming it `DeclaringType` then?

Reply:
Because a node can be something else than a field or property.

Reply:
So what do you think?

Reply:
@jnyrup ?

Reply:
Now the penny finally dropped, I was missing that the `ParentType` can also be e.g. a collection.